### PR TITLE
[Eventhub] Added Cluster commands and trusted_service_access_enabled parameter for Networkruleset

### PR DESCRIPTION
**Pull request by v-Ajnava**
…

**Description**  
Added new Cluster commands and trusted_service_access_enabled parameter for Networkruleset in az eventhubs namespace create/update commands

**Testing Guide**  
### Create cluster

> az eventhubs cluster create
example: 
az eventhubs cluster create --resource-group myresourcegroup --name mycluster --location mylocation --capacity 1 --tags tag=value

### Show cluster

> az eventhubs cluster show
example: 
az eventhubs cluster show --resource-group myresourcegroup --name mycluster  --tags tag=value

### Update cluster tags

> az eventhubs cluster update
example: 
az eventhubs cluster update --resource-group myresourcegroup --name mycluster

### Pre-provisioning cluster available in regions

> az eventhubs cluster available-region
example: 
az eventhubs cluster available-region

### List of Namespaces in cluster

> az eventhubs cluster namespace list
example: 
az eventhubs cluster namespace list --resource-group myresourcegroup --name mycluster

### List of cluster in resource group

> az eventhubs cluster list
example: 
az eventhubs cluster list --resource-group myresourcegroup

**History Notes**  

[eventhubs 1] Added new cluster commands , az eventhubs cluster
[eventhubs 2] az eventhubs namespace: added --enable-trusted-service-access parameter for newtworkruleset 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Eventhub

File: src/azure-cli/azure/cli/command_modules/eventhubs/_help.py [hunk 1]
```diff
@@ -425,3 +425,69 @@
...
+type: command
+short-summary: Create EventHubs Cluster
+examples:
+  - name: Update a new namespace.
+    text: az eventhubs cluster create --resource-group myresourcegroup --name mycluster --location mylocation --capacity 1 --tags tag=value
+"""
+
+helps['eventhubs cluster update'] = """
+type: command
+short-summary: Updates tags of EventHubs Cluster
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/_help.py:444]: Updates-> Update

File: src/azure-cli/azure/cli/command_modules/eventhubs/_help.py [hunk 2]
```diff
@@ -425,3 +425,69 @@
...
+type: command
+short-summary: List the quantity of available pre-provisioned Event Hubs Clusters, indexed by Azure region.
+examples:
+  - name: List of available pre-provisioned Event Hubs Clusters, indexed by Azure region.
+    text: az eventhubs cluster available-region
+"""
+
+helps['eventhubs cluster show'] = """
+type: command
+short-summary: Gets the resource description of the specified Event Hubs Cluster.
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/_help.py:460]: Gets -> Get

File: src/azure-cli/azure/cli/command_modules/eventhubs/_help.py [hunk 3]
```diff
@@ -425,3 +425,69 @@
...
+examples:
+  - name: Gets the resource description of the specified Event Hubs Cluster.
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/_help.py:462]: same

File: src/azure-cli/azure/cli/command_modules/eventhubs/_help.py [hunk 4]
```diff
@@ -425,3 +425,69 @@
...
+    text: az eventhubs cluster show --resource-group myresourcegroup --name mycluster
+"""
+
+helps['eventhubs cluster delete'] = """
+type: command
+short-summary: Deletes an existing Event Hubs Cluster.
+examples:
+  - name: Deletes an existing Event Hubs Cluster.
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/_help.py:470]: Delete

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py [hunk 5]
```diff
@@ -60,6 +62,19 @@ def load_arguments_eh(self, _):
...
                    help='The version of the KeyVault key to use.')
 
+# Cluster region
+    for scope in ['eventhubs cluster', 'eventhubs cluster namespace list']:
+        with self.argument_context(scope) as c:
+            c.argument('cluster_name', arg_type=name_type, id_part=None, help='Name of Cluster')
+
+    with self.argument_context('eventhubs cluster create') as c:
+        c.argument('location', arg_type=get_location_type(self.cli_ctx), id_part=None, help='Location of the Cluster, for locations of available pre-provision clusters, please check az evetnhubs ')
+        c.argument('capacity', type=int, help='Capacity for Sku')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/_params.py:72]: Is there a range?

File: src/azure-cli/azure/cli/command_modules/eventhubs/commands.py [hunk 6]
```diff
@@ -59,6 +65,16 @@ def load_command_table(self, _):
         g.command('delete', 'delete_authorization_rule')
         g.generic_update_command('update', getter_name='get_authorization_rule', setter_name='create_or_update_authorization_rule', custom_func_name='cli_autho_update')
 
+# Cluster Region
+    with self.command_group('eventhubs cluster', eh_clusters_util, resource_type=ResourceType.MGMT_EVENTHUB, client_factory=cluster_mgmt_client_factory) as g:
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/commands.py:69]: add min_api for command group if it is only supported in new api version

Comment (v-Ajnava) [reply src/azure-cli/azure/cli/command_modules/eventhubs/commands.py]: added min_api to cluster command group

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py (hunk 5 above)
Comment (v-Ajnava) [reply src/azure-cli/azure/cli/command_modules/eventhubs/_params.py]: allowed value is 1, in future might support more

File: src/azure-cli/azure/cli/command_modules/eventhubs/commands.py [hunk 7]
```diff
@@ -59,6 +65,16 @@ def load_command_table(self, _):
         g.command('delete', 'delete_authorization_rule')
         g.generic_update_command('update', getter_name='get_authorization_rule', setter_name='create_or_update_authorization_rule', custom_func_name='cli_autho_update')
 
+# Cluster Region
+    with self.command_group('eventhubs cluster', eh_clusters_util, resource_type=ResourceType.MGMT_EVENTHUB, client_factory=cluster_mgmt_client_factory, min_api='2018-01-01-preview') as g:
+        g.custom_command('create', 'cli_cluster_create')
+        g.show_command('show', 'get')
+        g.command('list', 'list_by_resource_group')
+        g.command('namespace list', 'list_namespaces')
+        g.command('delete', 'delete')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/commands.py:74]: prefer adding confirmation for delete command if possible

File: src/azure-cli/azure/cli/command_modules/eventhubs/tests/latest/test_cluster_commands.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        # update cluster
+        self.cmd('eventhubs cluster update --resource-group {rg} --name {clustername} --tags tag2=value2',
+                 checks=[self.check('tags', {'tag2': 'value2'})])
+
+        # Get cluster created in the resourcegroup
+        listclusterresult = self.cmd('eventhubs cluster list --resource-group {rg}').output
+        self.assertGreater(len(listclusterresult), 0)
+
+        # Delete cluster
+        # commented as the cluster can be deleted only after 4 hours
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/tests/latest/test_cluster_commands.py:65]: Add delete command

File: src/azure-cli/azure/cli/command_modules/eventhubs/custom.py [hunk 9]
```diff
@@ -99,6 +103,33 @@ def cli_namespace_list(cmd, client, resource_group_name=None):
...
+    ClusterSku = cmd.get_models('ClusterSku', resource_type=ResourceType.MGMT_EVENTHUB)
+
+    if cmd.supported_api_version(resource_type=ResourceType.MGMT_EVENTHUB, min_api='2018-01-01-preview'):
+        ehparam = Cluster()
+        ehparam.sku = ClusterSku()
+        ehparam.location = location
+        if not capacity:
+            ehparam.sku.capacity = 1
+        ehparam.tags = tags
+        cluster_result = client.create_or_update(
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/custom.py:118]: Add sdk_no_wait() here, e.g. https://github.com/Azure/azure-cli/blob/63d526fe74db2cfc882fbd344ac6c80d7dccb69e/src/azure-cli/azure/cli/command_modules/network/custom.py#L263

File: src/azure-cli/azure/cli/command_modules/eventhubs/tests/latest/test_cluster_commands.py [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+            'namespacename': self.create_random_name(prefix='eventhubs-nscli', length=20),
+            'tags': {'tag1=value1'},
+            'tags2': {'tag2=value2'},
+            'capacity': 1,
+            'sku': 'Standard',
+            'tier': 'Standard'
+        })
+
+        # create Resource group
+        self.cmd('az group create --resource-group {rg} --location {loc}')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/eventhubs/tests/latest/test_cluster_commands.py:35]: use ResourceGroupPreparer() to create resource group as  https://github.com/Azure/azure-cli/blob/daa9d5d6807a63abba5c7a7e82e98645422ef9a1/src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_account_scenarios.py#L1146

File: src/azure-cli/azure/cli/command_modules/eventhubs/tests/latest/test_cluster_commands.py (hunk 8 above)
Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/eventhubs/tests/latest/test_cluster_commands.py:65]: discussed online and will think another method for delete

**Review by Juliehzl — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py (hunk 5 above)
Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/eventhubs/_params.py]: please add arg_type=get_enum_types[1]

**Pull request merged**